index.bs

```diff
@@ -1592,36 +1621,71 @@ Constructors</h4>
-					in {{AudioContextOptions/latencyHint}}.
+			1. Let 
+				<dfn attribute for="AudioContext">[[pending resume promises]]</dfn> be
+				a slot on |context|, that is an initially empty ordered list of
+				promises.
 
-				2. If <code>contextOptions.{{AudioContextOptions/sampleRate}}</code> is specified,
-					set the {{BaseAudioContext/sampleRate}}
-					of this {{AudioContext}} to this value. Otherwise, use
-					the sample rate of the default output device. If the
-					selected sample rate differs from the sample rate of the
-					output device, this {{AudioContext}} MUST resample the
-					audio output to match the sample rate of the output device.
+			1. If <code>contextOptions</code> is given, perform the following
+				substeps:
+
+				1. If {{AudioContextOptions/sinkId}} is specified, set |sinkId| to
+					<code>contextOptions.{{AudioContextOptions/sinkId}}</code> and run
+					the following substeps:
+
+					1. If |document| is not allowed to use the feature identified by
+						<code>“speaker-selection”</code>, |sinkId| is the empty string,
+						or |sinkId| does not match any audio output device identified by
+						the result that would be provided by
+						{{MediaDevices/enumerateDevices()}}, abort these substeps.
+					
+					1. Set {{AudioContext/[[sink ID]]}} to |sinkId|.
+
+				1. Set the internal latency of |context| according to
+					<code>contextOptions.{{AudioContextOptions/latencyHint}}</code>, as
+					described in {{AudioContextOptions/latencyHint}}.
+
+				1. If <code>contextOptions.{{AudioContextOptions/sampleRate}}</code> is
+					specified, set the {{BaseAudioContext/sampleRate}} of |context| to
+					this value. Otherwise, use the sample rate of the default output
+					device. If the selected sample rate differs from the sample rate of
+					the output device, the user agent MUST resample the audio output to
+					match the sample rate of the output device.
+
+					Note: If resampling is required, the latency of |context|	may be
+					affected, possibly by a large amount.
+
+			1. Run the following substeps to prepare an audio output device for
+					rendering:
+	
+				1. If {{AudioContext/[[sink ID]]}} is
+					<code>"{{AudioContextSinkCategory/none}}"</code>, acquire system
+					resources to render the audio graph without an actual audio output
+					device. Abort these substeps.
 
-					Note: If resampling is required, the latency of the
-					AudioContext may be affected, possibly by a large
-					amount.
+				1. If {{AudioContext/[[sink ID]]}} is the empty string, acquire system
+					resources to render the audio graph with a default audio output
+					device. Abort these substeps.
 
-			5. If the context is <a>allowed to start</a>, send a
-				<a>control message</a> to start processing.
+				1. If {{AudioContext/[[sink ID]]}} is not the empty string, acquire
+					system resources to render the audio graph with the underlying audio
+					output device identified by {{AudioContext/[[sink ID]]}}.
```

Comment:
"Acquiring system resources" historically happens when starting, not when initializing. It's asynchronous to the `AudioContext` construction. It's lazy, so to speak. I think those steps are best spelled out in the control message part.

Reply:
Agreed. Done.